Mathlib/CategoryTheory/Bicategory/Grothendieck.lean

```diff
@@ -117,6 +118,148 @@ def forget : ∫ F ⥤ 𝒮 where
+      simp only [← Functor.map_comp_assoc, ← NatTrans.naturality_assoc]
+      simp only [Cat.id_obj, Cat.id_map, id_comp, eqToHom_naturality_assoc, Iso.inv_hom_id_app,
+        comp_id, assoc]
+      slice_rhs 3 5 => equals (F.map _).map ((F.mapId _).hom.app _) =>
+        symm
+        rw [conj_eqToHom_iff_heq']
+        congr
+        simp
+      rw [← NatIso.app_inv, ← Functor.map_comp]
+      simp
+
+/-- Making the equality of functors into an isomorphism. Note: we should avoid equality of functors
+if possible, and we should prefer `mapCompIso` to `map_comp_eq` whenever we can. -/
+def mapIdIso : map (𝟙 F) ≅ 𝟭 (∫ F) := eqToIso map_id_eq
+
+variable (α)
+
+variable {H : Pseudofunctor (LocallyDiscrete 𝒮ᵒᵖ) Cat.{v₂, u₂}}
+set_option maxHeartbeats 225000 in -- We need this for the second-to last simp to work.
+/-- The construction `map` strictly commutes with functor composition. -/
+theorem map_comp_eq (β : G ⟶ H) : map (α ≫ β) = map α ⋙ map β := by
+  fapply Functor.ext
+  · intro x
+    rfl
+  · intro X Y f
+    ext
+    · simp
+    · dsimp
+      simp only [Strict.associator_eqToIso, eqToIso_refl, Iso.refl_inv, Cat.id_app, Cat.comp_obj,
+        Iso.refl_hom, comp_id, id_comp, map_comp, assoc, H.mapComp_id_left_inv,
+        Strict.leftUnitor_eqToIso, eqToIso.inv, PrelaxFunctor.map₂_eqToHom, eqToHom_naturality,
+        Cat.comp_app, Cat.eqToHom_app, Cat.whiskerRight_app, Cat.id_obj, H.mapComp_id_right_inv,
+        Strict.rightUnitor_eqToIso, Cat.whiskerLeft_app, eqToHom_trans_assoc]
+      rw [eqToHom_map]
+      slice_rhs 6 8 => equals (H.map (𝟙 ⟨op X.base⟩)).map <| (H.map f.base.op.toLoc).map <|
+        (H.mapId ⟨op Y.base⟩).hom.app _ =>
+        symm
+        rw [conj_eqToHom_iff_heq']
+        congr 1 <;> simp only [Cat.id_obj, id_comp]
+        congr
+        simp
+      simp only [← Cat.whiskerLeft_app, ← Cat.whiskerRight_app, ← Cat.whiskerLeft_app,
+        ← NatTrans.comp_app_assoc, ← NatTrans.comp_app, ← Functor.comp_map]
+      slice_rhs 1 2 => erw [← NatTrans.naturality]
+      simp only [← Cat.whiskerLeft_app, ← Cat.whiskerRight_app, ← Cat.whiskerLeft_app,
+        ← NatTrans.comp_app_assoc, ← NatTrans.comp_app, ← Functor.comp_map]
+      simp only [Functor.comp_map, toOplax_toPrelaxFunctor, NatTrans.comp_app, Cat.comp_obj,
+        Cat.whiskerRight_app, Cat.whiskerLeft_app, Cat.id_obj, Cat.comp_map, Cat.id_map,
+        whisker_assoc, Strict.associator_eqToIso, eqToIso_refl, Iso.refl_hom, Iso.refl_inv, id_comp,
+        Bicategory.whiskerLeft_comp, assoc, comp_whiskerRight, Cat.id_app, map_id,
+        NatTrans.naturality, NatTrans.naturality_assoc, Iso.inv_hom_id_app_assoc]
+      congr
+      rw [← NatIso.app_inv, ← Functor.map_comp]
+      simp
+
+/-- Making the equality of functors into an isomorphism. Note: we should avoid equality of functors
+if possible, and we should prefer `mapCompIso` to `map_comp_eq` whenever we can. -/
+def mapCompIso (β : G ⟶ H) : map (α ≫ β) ≅ map α ⋙ map β := eqToIso (map_comp_eq α β)
```

Comment:
I would suggest the following structure for the code:
```lean
@[simps hom_app inv_app]
def mapCompIso (β : G ⟶ H) : map (α ≫ β) ≅ map α ⋙ map β :=
  NatIso.ofComponents (fun x ↦ Iso.refl _) (fun {x y} f ↦ by
    ext
    · simp
    · sorry)

lemma map_comp_eq (β : G ⟶ H) : map (α ≫ β) = map α ⋙ map β :=
  Functor.ext_of_iso (mapCompIso α β) (fun _ ↦ rfl) (fun _ ↦ by simp)
```
(Yes, it is possible to prove two functors are equal *after* we have shown they are isomorphic!)
The ingredients in the proof should be similar, the main difference is that there are no longer any `eqToIso`, and we directly get suitable simp lemmas.
In general, I would suggest proceeding in this way. You may first attempt to do the same for `mapIdIso`.